test/test_filters.cpp

```diff
@@ -516,6 +516,65 @@ BOOST_AUTO_TEST_CASE(complex_resampler) {
   
 }
 
+BOOST_AUTO_TEST_CASE(simple_resample) {
+
+  // time ranges
+  time_t origin(1643616000);
+  time_t ONE_DAY(60*60*24);
+
+  // data set
+  DbPointRecord::_sp db(new SqlitePointRecord());
+  db->setConnectionString("sample_dataset_2.db");
+  Clock::_sp c5m(new Clock(60*5, 0));
+
+  TimeSeries::_sp raw(new TimeSeries("raw", TSF_FOOT));
+  raw->setRecord(db);
+  raw->insertPoints({ Point(1643616049,14.703218460083008),
+    Point(1643616109,14.708871841430664),
+    Point(1643616169,14.711383819580078),
+    Point(1643616229,14.713897705078125),
+    Point(1643616349,14.719551086425781),
+    Point(1643616409,14.716409683227539),
+    Point(1643616469,14.708871841430664),
+    Point(1643616529,14.716409683227539),
+    Point(1643616589,14.718921661376953),
+    Point(1643616649,14.73274040222168),
+    Point(1643616769,14.708244323730469),
+    Point(1643616829,14.722063064575195),
+    Point(1643616889,14.727716445922852),
+    Point(1643616949,14.727087020874023),
+    Point(1643617009,14.727716445922852),
+    Point(1643617069,14.721433639526367),
+    Point(1643617129,14.724575042724609),
+    Point(1643617189,14.725202560424805)
+  });
+
+  TimeSeriesFilter::_sp rs(new TimeSeriesFilter());
+  rs->resample(ResampleModeLinear)->source(raw)->c(c5m)->units(TSF_FOOT)->name("resample");
+
+  auto point = rs->pointAtOrBefore(1643617189 + ONE_DAY);
+
+  BOOST_CHECK_EQUAL(point.isValid, true);
+  BOOST_CHECK_EQUAL(point.time, rs->clock()->timeBefore(1643617189));
+
+  auto point2 = rs->pointAtOrBefore(1643617189);
+  BOOST_CHECK_EQUAL(point.isValid, true);
+  BOOST_CHECK_EQUAL(point.time, rs->clock()->timeBefore(1643617189));
+
+  TimeSeriesFilter::_sp step(new TimeSeriesFilter());
+  step->resample(ResampleModeStep)->source(raw)->c(c5m)->units(TSF_FOOT)->name("step");
+
+  auto stepRs = step->pointAtOrBefore(1643617189 + ONE_DAY);
+  BOOST_CHECK_EQUAL(stepRs.isValid, true);
+  BOOST_CHECK_EQUAL(stepRs.time, rs->clock()->timeBefore(1643617189 + ONE_DAY));
+  BOOST_CHECK_EQUAL(stepRs.value, 14.725202560424805);
+
+  auto stepRs2 = step->pointAtOrBefore(1643617189);
+  BOOST_CHECK_EQUAL(stepRs2.isValid, true);
+  BOOST_CHECK_EQUAL(stepRs2.time, rs->clock()->timeBefore(1643617189) );
+  BOOST_CHECK_EQUAL(stepRs2.value, raw->pointAtOrBefore(rs->clock()->timeBefore(1643617189)).value);
```

Comment:
these tests should be updated as per the comment above, respecting the difference between "before" and "atOrBefore"